src/lib/pickles/wrap_verifier.ml

```diff
@@ -853,224 +853,243 @@ struct
+              absorb sponge Without_degree_bound z
+            in
+            let absorb_sorted_2_to_4 () =
+              let (_ :: rest) = l.sorted in
+              Vector.iter rest ~f:(fun z ->
+                  let z = Array.map z ~f:(fun z -> (Boolean.true_, z)) in
+                  absorb sponge Without_degree_bound z )
+            in
+            let absorb_sorted_5 () =
+              match l.sorted_5th_column with
+              | None ->
+                  ()
+              | Maybe (b, z) ->
+                  let z = Array.map z ~f:(fun z -> (b, z)) in
+                  absorb sponge Without_degree_bound z
+              | Some z ->
+                  let z = Array.map z ~f:(fun z -> (Boolean.true_, z)) in
+                  absorb sponge Without_degree_bound z
+            in
+            match[@warning "-4"]
+              (m.lookup_table_comm, m.runtime_tables_selector)
+            with
+            | _ :: Some _ :: _, _ | _, Some _ ->
+                let joint_combiner = sample_scalar () in
+                absorb_sorted_1 sponge ;
+                absorb_sorted_2_to_4 () ;
+                absorb_sorted_5 () ;
+                joint_combiner
+            | _ :: None :: _, None ->
+                absorb_sorted_1 sponge ;
+                absorb_sorted_2_to_4 () ;
+                absorb_sorted_5 () ;
+                { inner = Field.zero }
+            | _ :: Maybe (b1, _) :: _, Maybe (b2, _) ->
+                let b = Boolean.(b1 ||| b2) in
+                let sponge2 = Opt.copy sponge in
+                let joint_combiner_if_true =
+                  let joint_combiner = sample_scalar () in
+                  absorb_sorted_1 sponge ; joint_combiner
+                in
+                let joint_combiner_if_false : Scalar_challenge.t =
+                  absorb_sorted_1 sponge2 ; { inner = Field.zero }
+                in
+                Opt.recombine b ~original_sponge:sponge2 sponge ;
+                absorb_sorted_2_to_4 () ;
+                absorb_sorted_5 () ;
+                { inner =
+                    Field.if_ b ~then_:joint_combiner_if_true.inner
+                      ~else_:joint_combiner_if_false.inner
+                }
+            | _ :: Maybe (b, _) :: _, _ | _, Maybe (b, _) ->
+                let sponge2 = Opt.copy sponge in
+                let joint_combiner_if_true =
+                  let joint_combiner = sample_scalar () in
+                  absorb_sorted_1 sponge ; joint_combiner
+                in
+                let joint_combiner_if_false : Scalar_challenge.t =
+                  absorb_sorted_1 sponge2 ; { inner = Field.zero }
+                in
+                Opt.recombine b ~original_sponge:sponge2 sponge ;
+                absorb_sorted_2_to_4 () ;
+                absorb_sorted_5 () ;
+                { inner =
+                    Field.if_ b ~then_:joint_combiner_if_true.inner
+                      ~else_:joint_combiner_if_false.inner
+                }
```

Comment:
[minor] We should probably define a single function for the code between line 911-925 or 895-909 and use it twice.